app/controllers/pensions/lifetimeAllowance/AnnualAndLifetimeAllowanceCYAViewHelper.scala

```diff
@@ -0,0 +1,243 @@
+    lifetimeAllowancesViewModel.aboveLifetimeAllowanceQuestion.filter(_ == true).flatMap { _ =>
+      annualAllowancesViewModel.reducedAnnualAllowanceQuestion.filter(_ == true)
+        //TODO needs to be updated to match the prototype answer should either be the amount or no
+        .map(_ =>
+          annualAllowancesViewModel.aboveAnnualAllowanceQuestion match {
+            case Some(true) if annualAllowancesViewModel.aboveAnnualAllowance.isDefined =>
+              summaryListRowWithOptionalAmountValue(
+                "lifetimeAllowance.cya.aboveAnnualAllowance",
+                annualAllowancesViewModel.aboveAnnualAllowance,
+                annualRoutes.AboveReducedAnnualAllowanceController.show(taxYear))(messages)
+            case _ =>
+              summaryListRowWithBooleanValue(
+                "lifetimeAllowance.cya.aboveAnnualAllowance",
+                annualAllowancesViewModel.aboveAnnualAllowanceQuestion,
+                annualRoutes.AboveReducedAnnualAllowanceController.show(taxYear))(messages)
+          }
+        )
+    }
+  }
+
+  private def summaryRowForAnnualAllowanceTax(lifetimeAllowancesViewModel: PensionLifetimeAllowancesViewModel, annualAllowancesViewModel: PensionAnnualAllowancesViewModel, taxYear: Int)(implicit messages: Messages): Option[SummaryListRow] = {
+    lifetimeAllowancesViewModel.aboveLifetimeAllowanceQuestion.filter(_ == true).flatMap { _ =>
+      annualAllowancesViewModel.aboveAnnualAllowanceQuestion.filter(_ == true).flatMap( _ =>
+        annualAllowancesViewModel.reducedAnnualAllowanceQuestion.filter(_ == true)
+        .map(_ =>
+          annualAllowancesViewModel.pensionProvidePaidAnnualAllowanceQuestion match {
+            case Some(true) if annualAllowancesViewModel.taxPaidByPensionProvider.isDefined =>
+              summaryListRowWithOptionalAmountValue(
+                "lifetimeAllowance.cya.annualAllowanceTax",
+                annualAllowancesViewModel.taxPaidByPensionProvider,
+                routes.PensionProviderPaidTaxController.show(taxYear))(messages)
+            case _ =>
+              summaryListRowWithBooleanValue(
+                "lifetimeAllowance.cya.annualAllowanceTax",
+                annualAllowancesViewModel.pensionProvidePaidAnnualAllowanceQuestion,
+                routes.PensionProviderPaidTaxController.show(taxYear))(messages)
+          }
+        )
+      )
+    }
+  }
+
+  private def summaryRowForAnnualAllowanceSchemeTaxReferences(lifetimeAllowancesViewModel: PensionLifetimeAllowancesViewModel, annualAllowancesViewModel: PensionAnnualAllowancesViewModel, taxYear: Int)(implicit messages: Messages): Option[SummaryListRow] = {
+    lifetimeAllowancesViewModel.aboveLifetimeAllowanceQuestion.filter(_ == true).flatMap(_ =>
```

Comment:
Could you try a 'for comprehension' here?

Reply:
It doesn't look as clean imo